### PR TITLE
makewatch: a module to watch for make (or other long running) jobs

### PR DESCRIPTION
**Pull request by drwahl**
A module that will watch for (by default) make jobs and notify of their
status.  This can be used for other long-running processes by providing
an alternate 'name'.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "notmuch",
     "requests",
     "bs4",
-    "novaclient.v2"
+    "novaclient.v2",
+    "psutil", "getpass"
```
Comment (enkore) [inline docs/conf.py]: - getpass is part of the Python standard library
- psutil is already on the list (line 25)

File: i3pystatus/makewatch.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        cdict = {
+            "name": self.name,
+            "status": status
+        }
+
+        self.output = {
+            "full_text": self.format.format(**cdict),
+            "color": color
+        }
```
Comment (enkore) [inline i3pystatus/makewatch.py]: An option to hide it when idle would be nice

**Comment by drwahl** [comment]
I've been playing with getting the format to change dynamically (including "hiding"), but have not had success yet.  Are you able to provide me some pointers here?

**Comment by facetoe** [comment]
You can use the `on_leftclick`, `on_rightclick` callbacks to trigger actions in your module. Hiding could be as simple as changing the output to an empty string.

**Comment by enkore** [comment]
Cherry-picked relevant commits into 7be1361 8b8a3db b09e213

**Pull request closed without merging**